suite/tests/api/ir_regdeps.c

```diff
@@ -0,0 +1,325 @@
+/* **********************************************************
+ * Copyright (c) 2024 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include "configure.h"
+#include "dr_api.h"
+#include "tools.h"
+
+#define ASSERT(x)                                                                 \
+    ((void)((!(x)) ? (dr_fprintf(STDERR, "ASSERT FAILURE: %s:%d: %s\n", __FILE__, \
+                                 __LINE__, #x),                                   \
+                      dr_abort(), 0)                                              \
+                   : 0))
+
+#define ALIGN_BYTES 4
+
+static void
+test_instr_encode_decode_synthetic(void *dc, instr_t *instr)
+{
+    /* Encoded synthetic ISA instructions require 4 byte alignment.
+     * The largest synthetic encoded instruction has 16 bytes.
+     */
+    byte ALIGN_VAR(ALIGN_BYTES) bytes[16];
+
+    /* Convert a real ISA instruction to a synthetic ISA (DR_ISA_REGDEPS) instruction.
+     */
+    instr_t *instr_synthetic_converted = instr_create(dc);
+    instr_convert_to_isa_regdeps(dc, instr, instr_synthetic_converted);
+
```

Comment:
Add a check that the opcode is OP_INVALID (or should it be OP_UNDECODED?)

Reply:
Done. Checking for OP_INVALID, since that's the default for newly initialized instructions (OP_INVALID is 0).